packages/socket-mode/src/SocketModeClient.ts

```diff
@@ -30,6 +30,21 @@ enum State {
   Authenticated = 'authenticated',
 }
 
+/**
+ * Recursive definition for what a value might contain.
+ */
+interface Nesting {
+  [key: string]: NestedRecord | unknown;
+}
+
+/**
+ * Recursive definiton for possible JSON object values.
+ *
+ * FIXME: Prefer using a circular reference if allowed:
+ *   Record<string, NestedRecord> | NestedRecord[]
```

Comment:
Unfortunately this was tossing an error in the editor... Some [prior art was tried](https://stackoverflow.com/questions/59370450/typescript-type-alias-circularly-references-itself) but without much luck...

```
Type alias 'NestedRecord' circularly references itself.
```

Reply:
Fair enough, don't sweat it! We do [some seriously nasty gymnastics in deno-slack-sdk to avoid this](https://github.com/slackapi/deno-slack-sdk/blob/2b7b03d41cc18432f96759ab4c70c139b433e8fa/src/type_utils.ts#L1-L15); I _do not_ recommend it, though. Roll with the punches and move on!